web/packages/teleport/src/Bots/Edit/EditDialog.tsx

```diff
@@ -0,0 +1,285 @@
+                        value: r.name,
+                        label: r.name,
+                      }));
+                    }}
+                    noOptionsMessage={() => 'No roles found'}
+                    elevated={true}
+                  />
+                  <TraitsEditor
+                    css={css`
+                      padding-bottom: 16px;
+                    `}
+                    configuredTraits={
+                      selectedTraits ??
+                      data.traits
+                        .toSorted((a, b) => a.name.localeCompare(b.name))
+                        .map(t => ({
+                          traitKey: {
+                            value: t.name,
+                            label: t.name,
+                          },
+                          traitValues: t.values.toSorted().map(v => ({
+                            value: v,
+                            label: v,
+                          })),
+                        }))
+                    }
+                    setConfiguredTraits={setSelectedTraits}
+                    isLoading={false}
+                    label="Traits"
+                    addActionLabel="Add trait"
+                    addActionSubsequentLabel="Add another trait"
+                    autoFocus={false}
+                  />
+                  <FieldInput
+                    label="Max session duration"
+                    rule={requiredField('Max session duration is required')}
+                    value={
+                      selectedMaxSessionDuration ??
+                      formatDuration({
+                        seconds: data.max_session_ttl?.seconds ?? 43_200,
+                      })
+                    }
+                    onChange={e =>
+                      setSelectedMaxSessionDuration(e.target.value)
+                    }
+                    helperText={
+                      'A duration string such as 12h, 2h 45m, 43200s. Maximum is 24h. Valid time units are ns, us (or µs), ms, s, m, h.'
+                    }
+                  />
+                  <Alert kind="info" width={'100%'}>
+                    Updates to a bot&apos;s identity take affect when tbot next
+                    renews its certificates. By default, this happens every 20
+                    minutes.
+                  </Alert>
+                </>
+              ) : undefined}
+
+              {saveError ? (
+                <Alert kind="danger">Error: {saveError.message}</Alert>
+              ) : undefined}
+
+              {hasInconsistencies ? (
+                <Alert
+                  kind="warning"
+                  primaryAction={{ content: 'Dismiss', onClick: onCancel }}
+                >
+                  Warning: Some fields may not have updated correctly;{' '}
+                  {inconsistentFields.join(', ')}
+                </Alert>
+              ) : undefined}
+            </DialogContent>
+            <DialogFooter>
+              <ButtonPrimary
```

Comment:
I'd improve keyboard navigation a bit by allowing submitting the form with Enter. 
We need to wrap the fields and buttons in `<form>` and pass `onSubmit` there. Then we need to set `<ButtonPrimary type="submit">`.